docs/src/main/asciidoc/spring-cloud-commons.adoc

```diff
@@ -1007,6 +1013,18 @@ public class MyConfiguration {
 ----
 ====
 
+[[loadbalancer-lifecycle]]
+=== Spring Cloud LoadBalancer Lifecycle
+
+One type of bean that it may be useful to register using <<custom-loadbalancer-configuration,Custom LoadBalancer configuration>> is `LoadBalancerLifecycle`.
+
+The LoadBalancerLifecycle beans provide callback methods, named `onStart(Request<RC> request)` and `onComplete(CompletionContext<RES, T> completionContext)`, that you should implement to specify what actions should take place before and after load-balancing.
+
+`onStart(Request<RC> request)` takes a `Request` object as a parameter. It contains data that is used to select an appropriate instance, including the downstream client request and <<spring-cloud-loadbalancer-hints,hint>>. On the other hand, a `CompletionContext` object is provided to the `onComplete(CompletionContext<RES, T> completionContext)` method. It contains the LoadBalancer `Response`, including the selected service instance, the `Status` of the request executed against that service instance and (if available) the response returned to the downstream client, and (if an exception has occurred) the corresponding `Throwable`.
+
```

Comment:
Not sure where the right place to mention it is, but I was struggling to find descrptions of what `RC`, `RES`, and `T` represent/mean

Reply:
We could add a note after that paragraph to explain those three. To save Olga some typing, here's a structure for it:

NOTE: In the preceding method calls, `RC` means ..., `RES` means ..., and `T` means....

(Notice the extra period at the end, to close the sentence.)